distributed/stealing.py

```diff
@@ -415,4 +429,46 @@ def story(self, *keys):
         return out
 
 
+def _has_restrictions(ts):
+    """Determine whether the given task has restrictions and whether these
+    restrictions are strict.
+    """
+    return not ts.loose_restrictions and (
+        ts.host_restrictions or ts.worker_restrictions or ts.resource_restrictions
+    )
+
+
+def _can_steal(thief, ts, victim):
+    """Determine whether worker ``thief`` can steal task ``ts`` from worker
+    ``victim``.
+    """
+    if not _has_restrictions(ts):
+        return True
+
+    if (
+        ts.host_restrictions
+        and get_address_host(thief.address) not in ts.host_restrictions
+    ):
+        return False
+    elif ts.worker_restrictions and thief.address not in ts.worker_restrictions:
+        return False
+
+    return _has_resources(thief, victim.resources)
+
+
+def _has_resources(ws, required_resources):
```

Comment:
This could likely be inlined in `_can_steal`

Reply:
ok, I've inlined the function